src/evaluate/evaluator/text_classification.py

```diff
@@ -130,21 +130,28 @@ def compute(
         metric = self.prepare_metric(metric)
 
         # Compute predictions
+        start_time = perf_counter()
         predictions = pipe(data[input_column], truncation=True)
+        end_time = perf_counter()
```

Comment:
We should probably refactor this such that the pipe call is wrapped in the base class and we don't need to implement the perf logic for every task. 

In the parent class we would have something like this:
```py
def compute_predictions_with_perf(self):
    start_time = perf_counter()
    predictions = compute_predictions()
    end_time = perf_counter()
    return predictions, self._compute_time_perf(start_time, end_time, len(predictions))
```

And in the task specific class we define the logic for the task:
```py
def compute_predictions(self):
    return self.pipe(self.data[self.input_column], truncation=True)
```

Not 100% sure that's exactly the way to do it but I think this would make it easier for future additions. We don't need to tackle this in this PR necessarily but we should come up with a scalable solution next.

cc @fxmarty who's been working on other evaluators.

What do you think?

Reply:
I have no strong opinion on this, but I think a context manager could make sense here. Something like:

```python
with MeasureTime() as timing:
    # compute predictions here, not necessarily in a separate method?

results = {"latency": timing.latency, "throughput": timing.throughput}
```

I am not sure about having `compute()` in the parent class, how would custom cases be treated (like https://github.com/fxmarty/evaluate/blob/540956fb081cf909d29df02cb70dbab7db828b37/src/evaluate/evaluator/question_answering.py#L172-L178 )? Redefine `compute()` in the child class?

The PR looks good otherwise!